toolchain/check/testdata/class/adapter/adapt_copy.carbon

```diff
@@ -89,17 +84,35 @@ class AdaptNoncopyableIndirect {
 }
 
 fn H(a: AdaptNoncopyableIndirect) -> AdaptNoncopyableIndirect {
-  // CHECK:STDERR: fail_adapt_not_copyable_indirect.carbon:[[@LINE+4]]:37: error: cannot copy value of type `AdaptNoncopyableIndirect` [CopyOfUncopyableType]
+  // CHECK:STDERR: fail_adapt_not_copyable_indirect.carbon:[[@LINE+4]]:3: error: cannot copy value of type `Noncopyable` [CopyOfUncopyableType]
   // CHECK:STDERR:   var b: AdaptNoncopyableIndirect = a;
-  // CHECK:STDERR:                                     ^
+  // CHECK:STDERR:   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I think this is a better diagnostic, as it's more precise about what can not be copied: a particular type in the foundation tuple. But it's also a worse diagnostic as it ends up with a location that covers the whole statement (i think because the location becomes at the semicolon?). Would you think it's better if the location was still pointing at the original `value_id`? If yes, how would you accomplish that, by having a second LocId parameter in PerformBuiltinConversion?

Reply:
I think the ideal thing would be to add a note to this diagnostic saying "in copy of `AdaptNoncopyableIndirect`. We can achieve that by creating a `DiagnosticAnnotationScope` in the adapter conversion logic to add the note to any diagnostics created by the recursive call to `PerformBuiltinConversion`. (You can search for other uses of `DiagnosticAnnotationScope` to see how that's done elsewhere.)

Reply:
Thanks, done!